pkg/user/user_integration_test.go

```diff
@@ -0,0 +1,199 @@
+package user_test
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/dhis2-sre/im-manager/internal/middleware"
+	"github.com/dhis2-sre/im-manager/pkg/group"
+	"github.com/dhis2-sre/im-manager/pkg/inttest"
+	"github.com/dhis2-sre/im-manager/pkg/model"
+	"github.com/dhis2-sre/im-manager/pkg/token"
+	"github.com/dhis2-sre/im-manager/pkg/user"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUserHandler(t *testing.T) {
+	t.Parallel()
+
+	db := inttest.SetupDB(t)
+	userRepository := user.NewRepository(db)
+	userService := user.NewService(userRepository)
+	groupRepository := group.NewRepository(db)
+	groupService := group.NewService(groupRepository, userService)
+
+	err := createAdminUser("admin", "admin", userService, groupService)
+	require.NoError(t, err, "failed to create admin user and group")
+
+	authorization := middleware.NewAuthorization(userService)
+	privKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err, "failed to generate private key")
+	// TODO(ivo) I don't think we need/should use a pointer
+	authentication := middleware.NewAuthentication(&privKey.PublicKey, userService)
+	tokenSvc, err := token.NewService(tokenRepository{}, privKey, &privKey.PublicKey, 10, "secret", 10)
+	require.NoError(t, err)
+
+	client := inttest.SetupHTTPServer(t, func(engine *gin.Engine) {
+		userHandler := user.NewHandler(userService, tokenSvc)
+		user.Routes(engine, authentication, authorization, userHandler)
+	})
+
+	var user1ID string
+	{
```

Comment:
Subtests can be run individually (see go test regex). Creating users is tested but it's also a setup for further tests. So this has to run even if you only want to run one of the subtests below. If I would put this into a subtest you could run one of the subtests below only and it would not signup the needed user

Reply:
:+1: